collector/src/lib.rs

```diff
@@ -124,10 +124,9 @@ pub enum Error {
 }
 
 impl Error {
-    /// Construct an error from an HTTP response's status and problem details document, if present
-    /// in the body.
-    async fn from_http_response(response: Response) -> Error {
-        Error::Http(Box::new(HttpErrorResponse::from_response(response).await))
+    /// Construct an error from an HTTP error response.
+    fn from_http_error_response(http_error_response: HttpErrorResponse) -> Error {
```

Comment:
I'm not sure, but I suspect the only reason the previous version was a custom impl is because it was async. Now that this is an infallible, sync conversion from `HttpErrorResponse` to `Error`, can we make it `impl From<HttpErrorResponse> for Error`?